src/org/jitsi/impl/neomedia/MediaStreamImpl.java

```diff
@@ -3635,6 +3636,89 @@ public void injectPacket(RawPacket pkt, boolean data, TransformEngine after)
     }
 
     /**
+    * {@inheritDoc}
+    */
+    public boolean isKeyFrame(byte[] buf, int off, int len)
+    {
+        if (buf == null || buf.length < off + len)
+        {
+            return false;
+        }
+
+        Byte redPT = null, vp8PT = null, h264PT = null;
+        for (Map.Entry<Byte, MediaFormat> entry :
+            getDynamicRTPPayloadTypes().entrySet())
+        {
+            String encoding = entry.getValue().getEncoding();
+            Byte payloadType = entry.getKey();
+
+            if (Constants.RED.equalsIgnoreCase(encoding))
+            {
+                redPT = payloadType;
+            }
+            else if (Constants.VP8.equalsIgnoreCase(encoding))
+            {
+                vp8PT = payloadType;
+            }
+            else if (Constants.H264.equalsIgnoreCase(encoding))
+            {
+                h264PT = payloadType;
+            }
+        }
+
+        int payloadOff, payloadLen;
+        byte payloadType;
+
+        try {
+          // XXX this will not work correctly when RTX gets enabled!
+            if (redPT != null
+                && redPT == RawPacket.getPayloadType(buf, off, len))
+            {
+                REDBlock block = REDBlockIterator
+                    .getPrimaryBlock(buf,
+                        RawPacket.getPayloadOffset(buf, off, len),
+                        RawPacket.getPayloadLength(buf, off, len));
```

Comment:
Can we include this comment and adapt the code accordingly:

    // XXX There's RawPacket#getPayloadLength() but the implementation		
    // includes pkt.paddingSize at the time of this writing and		
    // we do not know whether that's going to stay that way.

In my opinion, eventually `RawPacket` has to go away. It needs to be split into different static classes and its methods need to do proper input validation. But again, that's what we have right now and we need to take that into consideration.